python/cuml/linear_model/logistic_regression.pyx

```diff
@@ -267,19 +278,54 @@ class LogisticRegression(Base,
 
     @generate_docstring()
     @cuml.internals.api_base_return_any(set_output_dtype=True)
-    def fit(self, X, y, convert_dtype=True) -> "LogisticRegression":
+    def fit(self, X, y, sample_weight=None,
+            convert_dtype=True) -> "LogisticRegression":
         """
         Fit the model with X and y.
 
         """
         # Converting y to device array here to use `unique` function
         # since calling input_to_cuml_array again in QN has no cost
         # Not needed to check dtype since qn class checks it already
-        y_m, _, _, _ = input_to_cuml_array(y)
-
+        y_m, n_rows, _, _ = input_to_cuml_array(y)
         self.classes_ = cp.unique(y_m)
         self._num_classes = len(self.classes_)
 
+        if self._num_classes == 2:
+            if self.classes_[0] != 0 or self.classes_[1] != 1:
+                msg = ("In binary classification,"
+                       "y should be filled with 0 and 1")
+                raise ValueError(msg)
+
+        if sample_weight is not None or self.class_weight_ is not None:
+            if sample_weight is None:
+                sample_weight = cp.ones(n_rows)
+
+            sample_weight, n_weights, D, _ = input_to_cuml_array(sample_weight)
+
+            if n_rows != n_weights or D != 1:
+                msg = "sample_weight should be of shape ({},)".format(n_rows)
+                raise ValueError(msg)
+
+            if self.class_weight_ is not None:
+                if self.class_weight_ == 'balanced':
+                    class_weight = n_rows / \
+                                   (self._num_classes *
+                                    cp.bincount(y_m.to_output('cupy')))
```

Comment:
It's better to use the `using_output_type("cupy")` context manager for this rather than calling `to_output` directly, like https://github.com/rapidsai/cuml/blob/bacb05e276c960c02b3d59934bbc48a47f76a17b/python/cuml/svm/svm_base.pyx#L313

Reply:
I couldn't make this work. I think that this only work with `CumlArrayDescriptor`.